.github/workflows/build.yml

```diff
@@ -28,12 +30,27 @@ jobs:
       - uses: actions/checkout@v4
       - uses: pnpm/action-setup@v2
         with:
-          version: 8
+          version: 10
       - uses: actions/setup-node@v4
         with:
-          node-version: "18"
+          node-version: "22"
           cache: "pnpm"
           cache-dependency-path: "frontend/pnpm-lock.yaml"
       - run: |
           pnpm install --no-frozen-lockfile
           pnpm run build
+
+  docker:
```

Comment:
`docker compose build` is a wrapper around `docker build`. Therefore, the `docker build` steps are not needed, when you call `docker compose build`. Or do I miss here something?